Doc for working with snapshots

allyhawkins opened the pull request:
### Which issue does this address?
Closes #262

### Briefly describe the scope of the added docs file, including whether you link out to any external docs.
<!-- If you are adding more than one docs file, how are they related to one another?-->
<!-- Are you adding any visual aids? Or, see next section if you think these docs would benefit from viz. -->

Here I added a doc on snapshots in LSfR. I talked about what a snapshot is and provide some guidance on how frequent to create them? I said once a week for now, but is that too much? 

I provided instructions on creating a snapshot from a VC or disk and then creating a new VC or disk from a snapshot. When creating snapshots I provided some guidance on naming the snapshot, mainly suggesting that they include a date in the snapshot name. 

I also included a section on adding a cost control rule if they create a new VC from a snapshot. 

### Will you need additional visual aids for these docs?
No

### Any other comments for reviewers?

While I was here I noticed we referenced when instances shut down slightly incorrectly based on the description in the cost control rule. So I made a small update to the sentence about that in creating VCs. 

I also added the last stub link 🎉 

### Author checklist

- [x] The docs page has been added to the `nav` section in `mkdocs.yml`
- [x] The docs adhere to the style guide [**structure directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#structure)
- [x] The docs adhere to the style guide [**language directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#structure)
- [x] The docs adhere to the style guide [**tone directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#tone)
- [x] The docs follow additional contributing instructions in [the docs contributing file](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/contributing.md)




### Reviewer checklist

Please refer to the [docs style guide](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md) and the [docs contributing guide](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/contributing.md) while reviewing this pull request.

- [ ] Do the docs adhere to the style guide [**structure directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#structure)?
- [ ] Do the docs adhere to the style guide [**language directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#structure)?
- [ ] Do the docs adhere to the style guide [**tone directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#tone)?
- [ ] Do the docs follow additional contributing instructions in [the docs contributing file](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/contributing.md)?

sjspielman commented on docs/software-platforms/aws/index.md:
```diff
@@ -30,4 +30,4 @@ Please refer to these pages to learn more about setting up and using LSfR:
 - [Working with LSfR volumes](./working-with-volumes.md)
 - [Developing analyses on LSfR](./starting-development-on-lsfr.md)
-- [Using snapshots on LSfR](STUB_LINK LSfR snapshots)
+- [Using snapshots on LSfR](./working-with-snapshots.md)
```
🎉

sjspielman commented on docs/software-platforms/aws/working-with-snapshots.md:
```diff
@@ -0,0 +1,114 @@
+
+## What is a snapshot
+
+Snapshots of both [virtual computers](./creating-vcs.md) and [disks](./working-with-volumes.md) can be created in Lightsail for Research (LSfR).
```
```suggestion
Snapshots of both [virtual computers](./creating-vcs.md) and [disks (i.e., volumes)](./working-with-volumes.md) can be created in Lightsail for Research (LSfR).
```

sjspielman replied:
Reading on, this is a general comment I suppose about terminology here.

sjspielman replied:
> Reading on, this is a general comment I suppose about terminology here.

nevermind this, i see we use "disk" next door in the other doc. Not a general comment!

allyhawkins replied:
Because we use the term "disk" throughout the doc on creating disks/ volumes, I think it's okay to just keep disk here.

sjspielman commented:
Noting I ran spell check locally and it all looks good 👍

sjspielman commented on docs/software-platforms/aws/working-with-snapshots.md:
```diff
@@ -0,0 +1,114 @@
+
+- Snapshots can be used to restore or create a new virtual computer or disk.
+This is particularly helpful if you need to restore a virtual computer or disk that has crashed for any reason.
+- When restoring a virtual computer from a snapshot, you can change the [instance size and select a different configuration for vCPUs and memory](./creating-vcs.md#choosing-an-instance).
```
While reviewing I wondered if this linkedsection should be called "choosing an instance size" instead?

sjspielman commented on docs/software-platforms/aws/working-with-snapshots.md:
```diff
@@ -0,0 +1,114 @@
+
+## Creating a new virtual computer or disk from a snapshot
+
+1. Find the previous virtual computer or disk that contains the snapshot you would like to use for creating a new virtual computer or disk.
```
```suggestion
1. Find the virtual computer or disk with the snapshot you would like to create a new virtual computer or disk from.
```

sjspielman commented on docs/software-platforms/aws/working-with-snapshots.md:
```diff
@@ -0,0 +1,114 @@
+
+    By default, the same instance size or disk size will be selected, but if you would like to re-size please choose a different option from the menu.
+
+    - Reference these [guidelines when choosing an instance size](./creating-vcs.md#choosing-an-instance).
```
```suggestion
    - Refer to these [guidelines when choosing an instance size](./creating-vcs.md#choosing-an-instance).
```

sjspielman commented on docs/software-platforms/aws/working-with-snapshots.md:
```diff
@@ -0,0 +1,114 @@
+    By default, the same instance size or disk size will be selected, but if you would like to re-size please choose a different option from the menu.
+
+    - Reference these [guidelines when choosing an instance size](./creating-vcs.md#choosing-an-instance).
+    - Remember that you will have a [monthly budget for any computational resources you use](../../getting-started/accessing-resources/getting-access-to-compute.md#monthly-budget).
```
```suggestion
    - Remember that you have a [monthly budget for any computational resources you use](../../getting-started/accessing-resources/getting-access-to-compute.md#monthly-budget).
```

sjspielman commented on docs/software-platforms/aws/working-with-snapshots.md:
```diff
@@ -0,0 +1,114 @@
+    </figure>
+
+1. Once you have chosen your size, select either `Create virtual computer` or `Create disk`.
+Your new virtual computer will be listed under the `Virtual computers` section and new disks can be found in the `Storage` section of LSfR.
```
```suggestion
New virtual computer will be listed under the `Virtual computers` section, and new disks will be in the `Storage` section of LSfR.
```

sjspielman commented on docs/software-platforms/aws/working-with-snapshots.md:
```diff
@@ -0,0 +1,114 @@
+## Creating a cost control rule
+
+If you have created a new virtual computer from a snapshot, you must add a cost control rule.
+When [creating a virtual computer](./creating-vcs.md) for the first time, a cost control rule is added, but this is not the case when creating a virtual computer from a snapshot.
```
```suggestion
Unlike when you [create a virtual computer from scratch](./creating-vcs.md), cost control rules are not automatically added when you create a virtual computer from a snapshot.
```

sjspielman commented on docs/software-platforms/aws/working-with-snapshots.md:
```diff
@@ -0,0 +1,114 @@
+1. Once you have chosen your size, select either `Create virtual computer` or `Create disk`.
+Your new virtual computer will be listed under the `Virtual computers` section and new disks can be found in the `Storage` section of LSfR.
+
+    If you are creating a new virtual computer, please be sure to follow the instructions below to [create a cost control rule](#creating-a-cost-control-rule).
+    A cost control rule ensures that your virtual computer does not continue to run when not actively in use, lowering your total cost.
```
I wonder if we should put this in a warning admonition? We don't want them to gloss over this

sjspielman reviewed (commented):
Overall things look good, mostly left phrasing comments!
I also added a comments about a section header. If you want to change that header, make sure to get the relative links to it too.

allyhawkins commented:
@sjspielman I incorporated your suggestions and updated the header for "Choosing an instance" to be "Choosing an instance size". I believe I got all the places we link to that section. 
I also added a warning box around the cost control information. 

This should be ready for another look.

sjspielman commented:
> I incorporated your suggestions and updated the header for "Choosing an instance" to be "Choosing an instance size". I believe I got all the places we link to that section.

Checked locally and confirmed!

sjspielman reviewed (approved):
time to cross the finish line!!
